server/loadConfigs.js

```diff
@@ -1,29 +1,79 @@
 const Config = require('bcfg');
+const fs = require('fs');
+const { resolve } = require('path');
 
-function loadConfigs(configs) {
-  // Setup config object
-  // bPanel specific configs, will be prefixed with `BPANEL_` in env
-  // dataDir defaults to `~/.[MODULE_NAME]`, where MODULE_NAME
-  // is the first argument passed to the Config constructor
-  // (i.e. 'bpanel' below)
+// load main bpanel config
+function loadMainConfig(options = {}) {
   const config = new Config('bpanel');
   config.load({
+    env: true,
     argv: true,
-    env: true
+    arg: true
   });
 
   // load any custom configs being passed in
-  config.inject(configs);
-
-  // inject client configs from a config in the client dir
-  // pass `client-id` as a command line arg
-  // or as env var BPANEL_CLIENT_ID
-  // client dir is also configurable with `client-dir`
-  // or `BPANEL_CLIENT_DIR`
-  const clientDir = config.str('clients-dir', 'clients');
-  const clientId = config.str('client-id', 'default');
-  config.open(`${clientDir}/${clientId}.conf`);
+  config.inject(options);
+
   return config;
 }
 
+/*
+ * Get an array of configs for each client
+ * in the module's home directory
+ * @param {Object} [configs] - Object of custom configs to inject
+ * into the bcfg config object
+ * @returns {Config[]} An array of Config object which can be
+ * used to load clients
+ */
+function loadConfigs(_config) {
+  // first let's load the parent bpanel config
+  let config = _config;
+
+  // if not passed bcfg object, create one
+  if (!(_config instanceof Config)) config = loadMainConfig(_config);
+
+  // clientsDir is the folder where all client configs
+  // should be saved and can be changed w/ custom configs
+  const clientsDir = config.str('clients-dir', 'clients');
+
+  // get full path to client configs relative to the project
+  // prefix which defaults to `~/.bpanel`
+  const clientsPath = resolve(config.prefix, clientsDir);
+  const clientFiles = fs.readdirSync(clientsPath);
+
+  // ignore file names that start with a '.' such as
+  // system files and files without `.conf` extension
+  // then load config for that client
+  return clientFiles
+    .filter(name => name[0] !== '.' && /.conf$/.test(name))
+    .map(fileName => {
+      // After filter, we load bcfg object for each client
+
+      // id is the file name without the extension
+      const clientId = fileName.slice(0, -5);
```

Comment:
What are these magic numbers?

Reply:
heh, `'.conf'.length`. I did feel a little icky about this. Which do you think is better 
```js
fileName.slice(0, -('.conf'.length))
``` 
or 
```js
const extLen = '.conf'.length;
const clientId = fileName.slice(0, -extLen);
``` 
or 
```js
// take off everything after the first extension marker (`.`), inclusive
const extIndex = fileName.indexOf('.');
const clientId = fileName.slice(0, extIndex);
```

Reply:
The last one seems the most flexible to me, but whats to stop somebody from having a file that has multiple `.`s?  There is a `lastIndexOf` right?

Maybe this is the safest solution: https://nodejs.org/api/path.html#path_path_parse_path

Reply:
my thought with taking the first one is that it's basically disallowing periods in the id completely which seems the safest to me but maybe that's not convenient if we're not explicit about it? path.parse could work though. The only thing that bothers me with it is I don't feel like we're parsing a path, but it does get the job done and it can confirm that the extension is `conf` too.